tests/timeseries/test_logical_meter.py

```diff
@@ -77,3 +77,69 @@ async def test_pv_power_no_pv_components(self, mocker: MockerFixture) -> None:
+
+            # Note: PvPowerFormula has a "nones-are-zero" rule, that says:
+            # * if the meter value is None, it should be treated as None.
+            # * for other components None is treated as 0.
+
+            expected_input_output: list[
+                tuple[list[float | None], list[float | None], Power | None]
+            ] = [
+                # ([meter_power], [pv_inverter_power], expected_power)
+                #
+                # Case 1: Both meters are available, so inverters are not used.
+                ([-1.0, -2.0], [None, -5.0], Power.from_watts(-3.0)),
+                ([-1.0, -2.0], [-10.0, -20.0], Power.from_watts(-3.0)),
+                # Case 2: The first meter is unavailable (None).
+                # Subscribe to the fallback inverter, but return None as the result,
+                # according to the "nones-are-zero" rule
+                ([None, -2.0], [-10.0, -20.0], None),
+                # Case 3: First meter is unavailable (None). Fallback inverter provides
+                # a value.
+                ([None, -2.0], [-10.0, -20.0], Power.from_watts(-12.0)),
+                ([None, -2.0], [-11.0, -20.0], Power.from_watts(-13.0)),
+                # Case 4: Both first meter and its fallback inverter are unavailable
+                # (None). Return 0 according to the "nones-are-zero" rule.
+                ([None, -2.0], [None, -20.0], Power.from_watts(-2.0)),
+                ([None, -2.0], [-11.0, -20.0], Power.from_watts(-13.0)),
+                # Case 5: Both meters are unavailable (None).
+                # Subscribe to the fallback inverter, but return None as the result,
+                # according "nones-are-zero" rule
+                ([None, None], [-5.0, -20.0], None),
+                # Case 6: Both meters are unavailable (None). Fallback inverter provides
+                # a values.
+                ([None, None], [-5.0, -20.0], Power.from_watts(-25.0)),
+                # Case 7: All components are unavailable (None).
+                # Return 0 according to the "nones-are-zero" rule.
+                ([None, None], [None, None], Power.from_watts(0.0)),
+                ([None, None], [-5.0, -20.0], Power.from_watts(-25.0)),
+                # Case 8: Meters becomes available and inverter values are not used.
+                ([-10.0, None], [-5.0, -20.0], Power.from_watts(-30.0)),
+                ([-10.0, -2.0], [-5.0, -20.0], Power.from_watts(-12.0)),
+            ]
+
+            for idx, (meter_power, pv_inverter_power, expected_power) in enumerate(
+                expected_input_output
+            ):
+                await mockgrid.mock_resampler.send_meter_power(meter_power)
+                await mockgrid.mock_resampler.send_pv_inverter_power(pv_inverter_power)
+                mockgrid.mock_resampler.next_ts()
+
+                result = await pv_power_receiver.receive()
+                assert result.value == expected_power, (
```

Comment:
```suggestion
                assert result.value.isclose(expected_power), (

Reply:
Why? Those values should be equal.

Reply:
This is because of float precision issues, in float arithmatic results.  We had issues of this sort and decided to only use `isclose` from now on, although we still have some old equal checks which need to be migrated.

`isclose` is an equality check in practice, because the default tolerance is very small.